src/coreclr/jit/lowerxarch.cpp

```diff
@@ -5800,7 +5802,32 @@ bool Lowering::TryGetContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode
 
     if (!node->OperIsHWIntrinsic())
     {
-        return supportsGeneralLoads && (IsContainableMemoryOp(node) || node->IsCnsNonZeroFltOrDbl());
+        bool canBeContained = false;
+
+        if (supportsGeneralLoads)
+        {
+            if (IsContainableMemoryOp(node))
+            {
+                // Code motion safety checks
+                //
+                if (transparentParentNode != nullptr)
+                {
+                    canBeContained = IsSafeToContainMem(containingNode, transparentParentNode, node);
```

Comment:
We don't need this for the `Load*` intrinsic cases below because its known to always be safe, is that right?

Reply:
I think so, yes. But not 100% sure.

By my understanding, for unary containing nodes, the safety check should immediately return safe, as the child's `gtNext` is the node, so there is nothing "in between" that can interfere. For higher arity operations the children can interfere and also there can be other interference in between the two from COMMA expansion.

Reply:
I'm going to add a quick early out to `IsSafeToContainMem` for the unary node case, so we are less tempted to optimize out the safety check call and possibly miss something.

Reply:
> By my understanding, for unary containing nodes, the safety check should immediately return safe, as the child's gtNext is the node, so there is nothing "in between" that can interfere

Do you mean that for unary operators `operand->gtNext == parent`? But that's not true for the linear order.
```scala
t1 = IND(addr)
     STOREIND(addr, ...) // Updates [addr] to t2
     UNARY_USER(t1) // Should observe t1, not t2.
```
Is valid LIR.

Reply:
Do you know of any way we'd create such a pattern? It would be nice to have more examples that fail if we mess this up.

At any rate, best not to be too clever here. With the added safety calls and follow-up check in `MakeSrcContained` we should have things covered, I hope.

Reply:
> Do you know of any way we'd create such a pattern?

Yep:
```cs
private static uint Problem(uint a, uint* b)
{
    uint zero = 0;
    return *b + Bmi2.MultiplyNoFlags(a, a, b) * zero;
}
```
```scala
N003 (???,???) [000015] ------------                 IL_OFFSET void   INLRT @ 0x002[E-] REG NA
N005 (  1,  1) [000003] -----------Z         t3 =    LCL_VAR   int    V01 arg1         u:1 edx REG edx $81
                                                  /--*  t3     int
N007 (  3,  2) [000004] *--XG-------         t4 = *  IND       int    REG eax <l:$1c1, c:$1c0>
N009 (  1,  1) [000005] ------------         t5 =    LCL_VAR   int    V00 arg0         u:1 ecx REG ecx $80
N011 (  1,  1) [000006] ------------         t6 =    LCL_VAR   int    V00 arg0         u:1 ecx (last use) REG ecx $80
N013 (  1,  1) [000007] -----------z         t7 =    LCL_VAR   int    V01 arg1         u:1 esi (last use) REG esi $81
                                                  /--*  t5     int
                                                  +--*  t6     int
                                                  +--*  t7     int
N015 (  4,  4) [000008] ---XG-------         t8 = *  HWINTRINSIC int     MultiplyNoFlags REG edx $83
                                                  /--*  t4     int
N017 ( 10,  9) [000012] ---XG-------              *  RETURN    int    REG NA $181

```